satellite/satellitedb/nodeselection.go

```diff
@@ -107,33 +104,43 @@ func (cache *overlaycache) selectStorageNodesOnce(ctx context.Context, reputable
 
 	var reputableNodeQuery, newNodeQuery partialQuery
 
+	aostClause := asOfSystemTimeClause{
+		interval:       criteria.AsOfSystemTimeInterval,
+		implementation: cache.db.implementation,
+	}
+	asOf := aostClause.getClause()
```

Comment:
Ideally this could be a single line something like:
```
asof := asOfSystemClause(cache.db.implementation, criteria.AsOfSystemTimeInterval)
```

Or maybe even:
```
asof := cache.db.AsOfSystemTimeClause(criteria.AsOfSystemTimeInterval)
```